### PR TITLE
Add default hierarchical context in group creation

### PR DESCRIPTION
**Pull request by aqurilla**
This fixes https://github.com/JabRef/jabref/issues/9141 by persisting the most recently used hierarchical context within `GroupPreferences`. This saved context is used when opening the `Add Group` dialog the next time.

image
image

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by HoussemNasri** [comment]
Thank you for working on this.

You're definitely going on the right track.

However, I believe it would be great if I can do this without having to open the "JabRef Preferences" dialog. One option that comes to mind is to set the "Hierarchical context" to the option selected in the last session. In other words, if the user selected "Intersection," then closed and reopened the "Add Group" dialog, "Intersection" should be still selected.

You can keep most of the code you have written so far. The changes to `GroupPreferences` look right on point, so you shouldn't need to touch any of it. Just undo changes to `GroupTab` and use `GroupPreferencesl#getDefaultHierarchicalContext` to set the default hierarchical context in `GroupDialogViewModel`.

**Comment by aqurilla** [comment]
Thanks for your feedback @HoussemNasri. I've updated the PR with the requested changes - the `GroupDialogViewModel` now persists the most recently used hierarchical context within `GroupPreferences`

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 ### Added
 
 - We added a field showing the BibTeX/biblatex source for added and deleted entries in the "External Changes Resolver" dialog. [#9509](https://github.com/JabRef/jabref/issues/9509)
+- We added an option in Groups Preferences to set the default hierarchical context used when creating a new group [#9141](https://github.com/JabRef/jabref/issues/9141)
```
Comment (HoussemNasri) [inline CHANGELOG.md:15]: Can you please update the changelog entry? And maybe put it under Changed instead of Added.

**Review by HoussemNasri — DISMISSED** [review]

**Review by Siedlerchr — APPROVED** [review]
Thank you very much!  +1 also for adding the test cases!
Looking forward towards more contributions :)

**Comment by aqurilla** [comment]
Thanks, happy to help!

**Pull request merged**